### PR TITLE
Optimize member set by using plain Go maps and parent-child chaining

### PR DESCRIPTION
**Pull request by turbolent**
## Description

Refactor the member set from using HAMT sets to Go map and parent-child chaining, just like in activations.

This improves performance of `BenchmarkCheckContractInterfaceFungibleTokenConformance` by ~8%.
______

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: runtime/interface.go [hunk 1]
```diff
@@ -202,10 +202,6 @@ func (i *emptyRuntimeInterface) RemoveAccountKey(_ Address, _ int) (publicKey []
 	return nil, nil
 }
 
-func (i *emptyRuntimeInterface) UpdateAccountCode(_ Address, _ []byte) error {
```
Comment (janezpodhostnik) [inline runtime/interface.go:205]: Not sure how this change connects to the rest of the changes.

Comment (turbolent) [reply runtime/interface.go:205]: Oh, this is unrelated. I saw it was old, unused code, then accidentally committed the change as part of this PR.

Comment (turbolent) [reply runtime/interface.go]: Reverted in 3ccad68

File: runtime/sema/member_set.go [hunk 2]
```diff
@@ -18,60 +18,94 @@
...
+}
+
+// Intersection returns a new set containing all members that exist in this and the given set.
+//
+func (ms *MemberSet) Intersection(otherMS *MemberSet) *MemberSet {
+
+	result := NewMemberSet(nil)
+
+	_ = ms.ForEach(func(member *Member) error {
+		if !otherMS.Contains(member) {
```
Comment (SupunS) [inline runtime/sema/member_set.go:94]: Just a minor suggestion: inverting the condition could reduce the number of return statements to  one.

**Review by SupunS — APPROVED** [review]

File: runtime/sema/member_set.go (hunk 2 above)
Comment (turbolent) [reply runtime/sema/member_set.go:94]: I usually try to structure code with early exits which avoids deep nesting. This also seems to be idiomatic / often seen in Go code.

**Pull request merged**